backend/oai.py

```diff
@@ -111,6 +114,43 @@ def ingest(self, from_date=None, metadataPrefix='oai_dc',
         records = self.client.listRecords(**args)
         self.process_records(records, metadataPrefix)
 
+    def load_base_dump(self, directory_path):
+        """
+        Given a path to a directory, representing an un-tar'ed BASE dump,
+        read all the bz2'ed files in it and save them to the database.
+
+        :param directory_path: the path to the directory where the BASE dump was un-tar'ed
+        """
+        metadata_prefix = 'base_dc'
+        generator = self.read_base_dump(directory_path, metadata_prefix)
+        self.process_records(generator, metadata_prefix, max_lookahead=10000)
+
+    def read_base_dump(self, directory_path, metadata_prefix):
+        """
+        Given a path to a directory, representing an un-tar'ed BASE dump,
+        read all the bz2'ed files in it as a generator of OAI records
+
+        :param directory_path: the path to the directory where the BASE dump was un-tar'ed
+        :param metadata_prefix: the metadata prefix to read the records
+        """
+        filenames = os.listdir(directory_path)
+        namespaces = self.client.getNamespaces()
+        metadata_registry = self.client.getMetadataRegistry()
+        for filename in filenames:
```

Comment:
Question here (same could apply to ORCID import actually): Wouldn't it be possible to replace the `for` loops on filenames by `parfor` loops (https://joblib.readthedocs.io/en/latest/parallel.html) to iterate over the files in parallel. Looks feasible (as files are independent), not sure about the expected processing gain here?

Reply:
Actually, we are already using a bit of parallel processing downstream in the import. That's not the case for ORCID however, I think.

Reply:
I am not sure about reading the files in parallel - I would assume DB access remains the bottleneck here, but I haven't actually checked that. It's true that decompressing and parsing large XML files has a cost.

Reply:
At the moment I expect the cost of implementation higher as the cost of not using `parfor`, so I leave as it is.